### PR TITLE
feat(developer): validate content file names in packages

### LINKED ISSUE #8754: chore(developer): SFilenameDoesNotFollowConventions_Message (CompilePackage.pas:143)

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #8754.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-package/src/compiler/package-validation.ts [hunk 1]
```diff
@@ -13,6 +13,10 @@ const MODEL_ID_PATTERN_PACKAGE = /^[a-z_][a-z0-9_]*\.[a-z_][a-z0-9_-]*\.[a-z_][a
 // const MODEL_ID_PATTERN_TS      = /^[a-z_][a-z0-9_]*\.[a-z_][a-z0-9_-]*\.[a-z_][a-z0-9_]*\.model\.ts$/;
 // const MODEL_ID_PATTERN_PROJECT = /^[a-z_][a-z0-9_]*\.[a-z_][a-z0-9_-]*\.[a-z_][a-z0-9_]*\.model\.kpj$/;
 
+// "Content files" within the package should adhere to these pattern:
+const CONTENT_FILE_BASENAME_PATTERN = /^[a-z0-9_+.-]+$/;
```
Comment (darcywong00) [inline developer/src/kmc-package/src/compiler/package-validation.ts:17]: Ah, I didn't realize '+' were valid filename characters. But the keyboard repo has
```
/legacy/v/visual_media_tamil_modular/source/help/visual_media_tamil_modular+chart.pdf
```

**Review by darcywong00 — APPROVED** [review]
lgtm

File: developer/src/kmc-package/src/compiler/package-validation.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc-package/src/compiler/package-validation.ts:17]: Indeed. There are many other legal chars and I cannot recall now why we chose to allow `+` but not, for example, `(`.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.106-alpha](https://keyman.com/downloads/releases/alpha/17.0.106)

**Pull request merged**